docs/src/rust-client/oracle_tutorial.md

```diff
@@ -0,0 +1,360 @@
+
+    let tx_increment_request = TransactionRequestBuilder::new()
+        .with_foreign_accounts(foreign_accounts)
+        .with_custom_script(tx_script)
+        .build()
+        .unwrap();
+
+    let tx_result = client
+        .new_transaction(oracle_reader_contract.id(), tx_increment_request)
+        .await
+        .unwrap();
+
+    let tx_id = tx_result.executed_transaction().id();
+    println!(
+        "View transaction on MidenScan: https://testnet.midenscan.com/tx/{:?}",
+        tx_id
+    );
+    // -------------------------------------------------------------------------
+    //  Submit transaction to the network
+    // -------------------------------------------------------------------------
+    let _ = client.submit_transaction(tx_result).await;
+
+    client.sync_state().await.unwrap();
+
+    Ok(())
+}
+```
+
+_Don't run this code just yet, we still need to create our smart contract that queries the oracle_
+
+In the code above, we specified the Pragma oracle account id `0x4f67e78643022e00000220d8997e33` and the BTC/USD pair `120195681`. The `get_oracle_foreign_accounts` function returns all of the `ForeignAccounts` that you will need to execute the transaction to get the price data from the oracle. Since Pragma's oracle depends on multiple publishers, this function queries all of the publisher account ids required to make a successful FPI call.
+
+To learn more about Pragma's oracle architecture, you can look at the source code here: https://github.com/astraly-labs/pragma-miden
+
+## Step 2: Build the price reader smart contract and script
+
+Just like in previous tutorials, for better code organization we will separate the Miden assembly code from our Rust code.
+
+Create a directory named `masm` at the **root** of your `miden-counter-contract` directory. This will contain our contract and script masm code.
+
+Initialize the `masm` directory:
+
+```bash
+mkdir -p masm/accounts masm/scripts
+```
+
+This will create:
+
+```
+masm/
+├── accounts/
+└── scripts/
+```
+
+### Oracle price reader smart contract
+
+Below is our oracle price reader contract. It has a a single exported procedure: `get_price`
+
+The import `miden::tx` contains the `tx::execute_foreign_procedure` which we will use to read the price from the oracle contract.
+
+#### Here's a breakdown of what the `get_price` procedure does:
+
+1. Pushes `0.0.0.120195681` onto the stack, representing the BTC/USD pair in the Pragma oracle.
+2. Pushes `0xb86237a8c9cd35acfef457e47282cc4da43df676df410c988eab93095d8fb3b9` onto the stack which is the procedure root of the `get_median` procedure in the oracle.
+3. Pushes `599064613630720.5721796415433354752` onto the stack which is the oracle id prefix and suffix.
+4. Calls `tx::execute_foreign_procedure` which calls the `get_median` procedure via foreign procedure invocation.
+
+Inside of the `masm/accounts/` directory, create the `oracle_reader.masm` file:
+
+```masm
+use.miden::tx
+
+# => []
+export.get_price
```

Comment:
Can you always add comments before the call, what this call does? 

I think that would be useful for educational services